src/blas/backends/cublas/cublas_task.hpp

```diff
@@ -33,6 +33,7 @@ static inline void host_task_internal(H &cgh, sycl::queue queue, F f) {
     cgh.host_task([f, queue](sycl::interop_handle ih) {
```

Comment:
I don't think it's necessarily an issue in practice but having a closer look at this I think capturing the queue like that is supposed to be undefined behavior.

The SYCL 2020 spec says in the section describing the host tasks ([Section 4.10.1](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:interfaces.hosttasks.overview)):
> However, capturing or using any SYCL class with reference semantics (see [Section 4.5.2](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:reference-semantics)) is undefined behavior.

Which includes the `queue` object, the correct way of doing this is supposed to be using the `interop_handle`, which provides `get_native_queue`, `get_native_context` and so on.

This could be an easy change for the most part aside from the `sycl::detail::pi::contextSetExtendedDeleter` that is used for the cublas handle, that works on a SYCL context which is not available through the interop handle.

Reply:
Isn't this just a by-copy capture? We would need `&queue` to capture by reference.